### PR TITLE
Fixed issue in `system-info.sh`regarding the parsing of `lscpu` output.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

When using `lscpu` to determine the CPU frequency, we want to preferentially use the `CPU max MHz` value over `CPU MHz`, as the later reflects the current CPU frequency in most cases, not the frequency that the CPU would be advertised with.

Unfortunately, the original code for parsing this info out of `lscpu` output did not take into account that some systems do not report a `CPU max MHz` value, but still report `CPU MHz`. Reporting something on such systems is better than reporting nothing, so we should fall back to `CPU MHz` if it's present.

##### Component Name

area/daemon

##### Test Plan

Verified locally to not change behavior for systems that weren't affected, and to fix it for systems that were.

##### Additional Information

Originally spotted by @manos-saratsis.

### CODE REVIEW

**Comment by thiagoftsm** [comment]
When I ran the script on terminal, I received the following error message:

```
daemon/system-info.sh: line 195: [:  MHz: integer expression expected
```

I tested it on `Slackware current` with `bash 5.0.16`.

**Comment by Ferroin** [comment]
> When I ran the script on terminal, I received the following error message:
> 
> ```
> daemon/system-info.sh: line 195: [:  MHz: integer expression expected
> ```
> 
> I tested it on `Slackware current` with `bash 5.0.16`.

Same here, I hadn't noticed this when testing because I was only looking at the CPU_FREQ line. I should have a fix shortly.

**Comment by thiagoftsm** [comment]
When I ran now on bare metal it worked fine, but when I started the VirtualBox and I tested the script inside it, I got the following output:

```
NETDATA_CONTAINER_OS_NAME=none
NETDATA_CONTAINER_OS_ID=none
NETDATA_CONTAINER_OS_ID_LIKE=none
NETDATA_CONTAINER_OS_VERSION=none
NETDATA_CONTAINER_OS_VERSION_ID=none
NETDATA_CONTAINER_OS_DETECTION=none
NETDATA_HOST_OS_NAME=Debian GNU/Linux
NETDATA_HOST_OS_ID=debian
NETDATA_HOST_OS_ID_LIKE=unknown
NETDATA_HOST_OS_VERSION=10 (buster)
NETDATA_HOST_OS_VERSION_ID=10
NETDATA_HOST_OS_DETECTION=/etc/os-release
NETDATA_SYSTEM_KERNEL_NAME=Linux
NETDATA_SYSTEM_KERNEL_VERSION=4.19.0-8-amd64
NETDATA_SYSTEM_ARCHITECTURE=x86_64
NETDATA_SYSTEM_VIRTUALIZATION=oracle
NETDATA_SYSTEM_VIRT_DETECTION=systemd-detect-virt
NETDATA_SYSTEM_CONTAINER=unknown
NETDATA_SYSTEM_CONTAINER_DETECTION=none
NETDATA_SYSTEM_CPU_LOGICAL_CPU_COUNT=1
NETDATA_SYSTEM_CPU_VENDOR=GenuineIntel
NETDATA_SYSTEM_CPU_MODEL=Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
NETDATA_SYSTEM_CPU_FREQ=0
NETDATA_SYSTEM_CPU_DETECTION=lscpu
NETDATA_SYSTEM_TOTAL_RAM=770088960
NETDATA_SYSTEM_RAM_DETECTION=procfs
NETDATA_SYSTEM_TOTAL_DISK_SIZE=16777216
NETDATA_SYSTEM_DISK_DETECTION=sysfs
```

But, when I run the `lscpu` direct on VM, I can see the CPU frequency:

```
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       39 bits physical, 48 bits virtual
CPU(s):              1
On-line CPU(s) list: 0
Thread(s) per core:  1
Core(s) per socket:  1
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               142
Model name:          Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
Stepping:            9
CPU MHz:             2903.998
BogoMIPS:            5807.99
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            4096K
NUMA node0 CPU(s):   0
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq monitor ssse3 cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase avx2 invpcid rdseed clflushopt flush_l1d
```

I tested on a VM with Debian 10.3.

File: daemon/system-info.sh [hunk 1]
```diff
@@ -191,7 +191,10 @@ if [ -n "${lscpu}" ] && lscpu >/dev/null 2>&1 ; then
         LCPU_COUNT="$(echo "${lscpu_output}" | grep "^CPU(s):" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
         CPU_VENDOR="$(echo "${lscpu_output}" | grep "^Vendor ID:" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
         CPU_MODEL="$(echo "${lscpu_output}" | grep "^Model name:" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
-        possible_cpu_freq="$(echo "${lscpu_output}" | grep -F "CPU max MHz:" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' | grep -o '^[0-9]*') MHz"
+        possible_cpu_freq="$(echo "${lscpu_output}" | grep -F "^CPU max MHz:" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' | grep -o '^[0-9]*') MHz"
+        if [ "${possible_cpu_freq}" = " MHz" ] ; then
+            possible_cpu_freq="$(echo "${lscpu_output}" | grep -F "^CPU MHz:" | cut -f 2 -d ':' | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' | grep -o '^[0-9]*') MHz"
+        fi
```
Comment (prologic) [inline daemon/system-info.sh:197]: Using `grep -F` here is an error. `-F` means a basic. Both of these will result in a non-match. What you want is either `-G` (_basic regex_) or `-E` (_extended regex_). You've used a `^...` in your pattern indicating a regex of some kind :)

**Comment by Ferroin** [comment]
@thiagoftsm Can you take another look at this? The issue you saw should be fixed now.

**Pull request merged**